### ISSUE TITLE
The 620/ directory contains information about *all* algorithms

### ISSUE THREAD

**Issue by bangerth** [ACM-TOMS/CALGO#41]
@TimHopkins The `0620/` directory contains files `0620/Original-Submission/620.bib` that has bibtex entries for all algorithms, and so does the file `0620/Original-Submission/DBase/DBase.txt`. The purpose of this is explained in `0620/Original-Submission/DBase/README.txt`: Algorithm 620 is meant as a catch-all article that references *all of the other* algorithms.

This no longer makes sense today: We keep this information up to date in the `Metadata` directories of all of the algorithms, and I think we should stop updating the contents of the directory. (In fact, I think we should roll back to only include everything up to and including 620 itself.)

Does this make sense? Opinions?

**Comment by pauldj** [comment]
I was going to discuss exactly this with Tim next week. I was wondering who is (painstakingly) updating 620; if it's all automated, we could automate the updating of 620 too. 
Where is the bibentry coming from? Should we have it in the Metadata/ directory of each algorithm, separately?

**Comment by bangerth** [comment]
I don't think we should duplicate information. We already have the means to create bib entries in `Metadata/` for each article, using the doi2bib service. Or we could split the existing file with bib entries and use those for the individual directories -- that's a quality control issue which way we want to go.

But I believe that we should generate collated lists (if that was desirable) automatically, from information stored within each of the subdirectories, rather than doing this by hand in a non-obvious place.

**Comment by pauldj** [comment]
Tim and I talked about this. 
- 620.zip is updated by hand by Tim. 
- It turns out that 620 contains a little bit more info than the sum of all the bibentries, as it includes dependencies for the remarks. In that respect, it would not make sense to undo his effort by rolling back to 620 itself.
- Tim finds it useful to have all the bibentries in one single file. This we could achieve ourselves with a script.

**Comment by bangerth** [comment]
So how do we move forward then? I'm not sure it is maintainable in the long run to update 620 by hand. My preference would be to have a script that can generate this kind of information from stuff we put into `Meta/` in each subdirectory.

**Comment by pauldj** [comment]
I agree about maintainability in the long run, and my preference is also a script as you describe. My plan is to keep talking to Tim, to better understand what the extra information actually consists of. For now, if he keeps updating 620, I'll keep pushing it into CALGO. I

**Comment by TimHopkins** [comment]
Regarding Algorithm 620:
========================

Although it is a little crude, the data file associated with Algorithm 620 (and updated with each update to the CALGO) is probably the most complete and accurate record we have of CALGO related articles that have appeared in CACM and TOMS.

I would agree that the Fortran code that accompanies it is obsolescent and irrelevant; this has been frozen for many years.

Most Remark and some Certification articles contain details of code changes that fix bugs, improve efficiency, etc. Having these accessible is important as they provide an audit trial for the software. 

As I’ve recently discovered many CALGO related articles are difficult/impossible to find in the ACM Digital Library.

This is mainly because the individual articles do not have their own doi and/or entry in the table of contents for the particular issue in which they appear. Hence the search engine has no means of locating them. In a small number of cases the pages containing relevant articles do not appear in the digital library at all.

Attempting to generate a definitive list of doi's for the GitHub project uncovered a large number of these problems which I am trying to get fixed.

I intend to fold the doi's I have for all the CALGO articles into the 620 data file and then I propose to freeze it, since any new articles should be easily dealt with. Does that sound like a reasonable plan?
